src/runtime/virtcontainers/pkg/oci/utils.go

```diff
@@ -442,6 +443,20 @@ func addHypervisorConfigOverrides(ocispec specs.Spec, config *vc.SandboxConfig)
 			config.HypervisorConfig.EntropySource = value
 		}
 	}
+	if epcSize, ok := ocispec.Annotations[vcAnnotations.SGXEPC]; ok {
+		quantity, err := resource.ParseQuantity(epcSize)
+		if err != nil {
+			return fmt.Errorf("Couldn't parse EPC '%v': %v", err, epcSize)
+		}
+
+		if quantity.Format != resource.BinarySI {
+			return fmt.Errorf("Unsupported EPC format '%v': use Ki | Mi | Gi | Ti | Pi | Ei as suffix", epcSize)
+		}
+
+		size, _ := quantity.AsInt64()
+
+		config.HypervisorConfig.EPCSize = int32(size)
```

Comment:
This looks odd to me: why are we parsing the value as a 64-bit int but then forcing it into a **signed** 32-bit int?

Reply:
because of clh API expects a signed 32-bit -  https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/virtcontainers/pkg/cloud-hypervisor/client/api/openapi.yaml#L436-L438

Reply:
Hmm. Now I'm *really* confused:

- https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/virtcontainers/pkg/cloud-hypervisor/client/api/openapi.yaml#L1155..L1162

Reply:
https://github.com/kata-containers/kata-containers/blob/740c45ac615dd0147abc19c76260bd7fc6a5b8d9/src/runtime/virtcontainers/pkg/cloud-hypervisor/client/model_sgx_epc_config.go#L12-L15

Reply:
This is just clearly wrong it is specified as u64 here - https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/virtcontainers/pkg/cloud-hypervisor/client/api/openapi.yaml#L1160

Obviously the generator is buggy.

Reply:
I have raised a PR to fix it https://github.com/cloud-hypervisor/cloud-hypervisor/pull/1760
Can we merge this and fix this [issue](https://github.com/kata-containers/kata-containers/issues/810) later (in the next release of cloud-hypervisor) ?

Reply:
I guess we may need to do this. However, do we have an ETA for when the next CLH release would be available?

Clearly, we cannot support exabyte sized EPC sections currently.

Hence, although there is a risk we'll forget to remove it, I'd prefer we add in a check to ensure that the size specified by the annotation *can* be dealt with here (fits in 32-bits). Also, we should really add a warning message (unconditionally?) if the annotation is found with a reference to https://github.com/kata-containers/kata-containers/issues/810 and explaining that support is currently only partial.

Reply:
> I guess we may need to do this. However, do we have an ETA for when the next CLH release would be available?

cc @likebreath @rbradford @sboeuf

> Hence, although there is a risk we'll forget to remove it, I'd prefer we add in a check to ensure that the size specified by the annotation can be dealt with here (fits in 32-bits). 

fail if it doesn't fit?

Also, we should really add a warning message (unconditionally?) if the annotation is found with a reference to #810 and explaining that support is currently only partial

sounds good, I can add it

Reply:
@devimc We make 6 weekly releases. Since you have your own copy of the .yaml file for your purposes I suggest you just copy the changed one in.

Reply:
thanks @rbradford - sounds good to me, @jodh-intel @likebreath wdyt?

Reply:
wfm ;)

Reply:
@devimc We don't have to wait clh next release to use its latest changes. You can just update the `version.yaml` with the `commit hash` for the `cloud-hypervisor` entry. I feel it will be cleaner. Otherwise, you will need to manually update the `cloud-hypervisor.yaml` file. Also, either way, we will need to re-generate the clh client code.